Add product

eid-sonkrot opened the pull request.

Commit:
In this commit, we implemented methods to allow the system to add products to the products list. We made changes in two classes: Inventory, where we added the AddProduct method, and UserInterface, where we added commands to support this functionality. Additionally, we fixed an error that was occurring in the Input method of the Product class.

Alatrash814 commented on Inventory Management System/Inventory.cs:
```diff
@@ -8,5 +8,29 @@ namespace Inventory_Management_System
     {
+        //List of products in our System 
+        private static List<Product> products;
+        public Inventory()
```
```suggestion
        private static List<Product> products;

        public Inventory()
```

Alatrash814 commented on Inventory Management System/Inventory.cs:
```diff
@@ -8,5 +8,29 @@ namespace Inventory_Management_System
     internal class Inventory
     {
+        //List of products in our System 
+        private static List<Product> products;
```
Please convert this to non-static

Alatrash814 replied:
```suggestion
        private List<Product> products = new List<Product>():
```

Alatrash814 commented on Inventory Management System/Inventory.cs:
```diff
@@ -8,5 +8,29 @@ namespace Inventory_Management_System
+        public Inventory()
+        {
+            products = new List<Product>();
+
```
Remove this empty line.

Alatrash814 commented on Inventory Management System/Inventory.cs:
```diff
@@ -8,5 +8,29 @@ namespace Inventory_Management_System
+
+        }
+        //add Product to Products List
+        public  static  void AddProduct( Product NewProduct)
```
```suggestion
        public void AddProduct( Product NewProduct)
```

Alatrash814 commented on Inventory Management System/Inventory.cs:
```diff
@@ -8,5 +8,29 @@ namespace Inventory_Management_System
+            }
+            else
+            {
+                Console.WriteLine("The product you want to add already exists." +
+                    " If you want to update the product information," +
+                    " please select that option from the main list.");
```
```suggestion
                Console.WriteLine(@"The product you want to add already exists.\n"
                    " If you want to update the product information,"
                    " please select that option from the main list.");
```

Alatrash814 commented on Inventory Management System/Inventory.cs:
```diff
@@ -8,5 +8,29 @@ namespace Inventory_Management_System
+                    " please select that option from the main list.");
+            }
+            System.Threading.Thread.Sleep(1000);
+
```
Please remove this empty line.

Alatrash814 commented on Inventory Management System/Product.cs:
```diff
@@ -25,11 +25,12 @@ public Product()
             bool IsValid = false;
-            
-           
+            string Name;
+            Double Price;
+            int Quantity;
```
```suggestion
            var name = (string?) null;
            var price = (double?) null;
            var quantity = (int?) null;
```

Alatrash814 commented on Inventory Management System/UserInterface.cs:
```diff
@@ -11,18 +11,23 @@ internal class UserInterface
         {
-            
 
+            Inventory inventory = new Inventory();
```
var
```suggestion
            var inventory = new Inventory();
```

Commit:
mearge with ProductClass branch

Commit:
Refactor AddProduct function to comply with style guidelines

In this commit, we refactored the AddProduct function to adhere to the style guidelines. The following updates were made:

1. Removed extra spaces for improved code formatting.
2. Converted the AddProduct function from a static to a non-static method, aligning with object-oriented principles.
3. Enhanced code readability by organizing the switch statement into a separate method and gracefully handling invalid input.

The changes were exclusively applied to the AddProduct function, resulting in a more maintainable and readable codebase, consistent with the specified style guidelines.

eid-sonkrot committed:
Refactor code style

eid-sonkrot committed:
Refactor code style

eid-sonkrot committed:
Merge branch 'master' into AddProduct

eid-sonkrot committed:
refactor code style

eid-sonkrot committed:
Refactor code Style
